### DISCUSSION TITLE
Comments - Most effective ways to push within GitHub Actions

### DISCUSSION THREAD

**Discussion by JohT** [JohT/johtizen#34] [General]
Contains the comments of the blog article [Most effective ways to push within GitHub Actions](https://joht.github.io/johtizen/build/2022/01/20/github-actions-push-into-repository.html).
❕**Note** that every post will be shown in the comments at the bottom of the blog article.

**Comment by vincerubinetti** [comment]
> actions/checkout@v3 behaves differently

Could you elaborate on that?

**Reply by JohT** [reply]
Hi @vincerubinetti,

as described further below in [GIT commit within a pull request - Adapt Checkout](https://joht.github.io/johtizen/build/2022/01/20/github-actions-push-into-repository.html#adapt-checkout), `actions/checkout@v3` needs the following `ref` parameter to be able to retrieve the author and the message of the last commit with `git log`, when the GitHub Actions event is of type "pull_request" in contrast to "push". So it behaves differently according to the GitHub Actions event type.

```
- uses: actions/checkout@v3
  with:
    ref: ${{ github.event.pull_request.head.ref }}
```

I'll see if i could add an link to the more detailed description since this is indeed misleading. Thanks for your comment!

**Comment by gregsdennis** [comment]
This is very comprehensive.  I'm struggling with pulling in a second repo, updating files, and then pushing those changes back to the secondary repo.  Any chance you've been able to get that working?  The only thing that I can't get working is the push.

**Reply by JohT** [reply]
Thank you. I never had the requirement to push to another repository. 

- I can imagine that it could make sense to create a new branch and then a pull request for it. If there are e.g. restrictions to push to main then this is needed anyway. 

- Maybe you can use tools like [Renovate](https://github.com/renovatebot/renovate) for that too. I'm not sure if there is a way to only use the pull request part of it or to extend it beyond version number updates. 

- Or you could question the need and bring both repositories together or other structural changes like that.
 
Just a couple of thoughts of mine.....

**Reply by gregsdennis** [reply]
Thanks, but it does make sense to keep them separate in my case.

I'm building a documentation site for my project, and it includes MSDN-style API docs that are generated from the libraries.  I'm trying to get them to generate when I check in changes to my code, then push the generated docs to the docs site repo.

I did eventually get it working with https://github.com/cpina/github-action-push-to-another-repository, though the first time I tried that, it didn't work.  Not sure why it worked the second time.

You can see what I'm doing in my action: https://github.com/gregsdennis/json-everything/blob/master/.github/workflows/update-docs.yml

**Comment by JohT** [comment]
I've added a ✨**new**✨ [section](https://joht.github.io/johtizen/build/2022/01/20/github-actions-push-into-repository.html#git-conflict-resolution) on how to resolve git conflicts automatically for anyone who encounters failing pipelines due to race conditions. Additionally, I've added a section and two links regarding the commit email address and how it can be set.
